### ISSUE TITLE
Consider Ignoring Type Annotation If add_argument is Used

### ISSUE THREAD

**Issue by Chris3606** [innot/argparseDecorator#8]
I've been using the changes in v1.3.1; they are working well, thank you!  I did notice one small annoyance though.

When writing code that uses `add_argument`, you can't use static type annotations at all on the corresponding argument in the function.  For example, the following code works:

```py
@cli.command
@cli.add_argument('--flag', type=argparse.FileType('r', encoding='latin-1'))
def read(flag):
    ...
```

However, if I try to add a type annotation to `--flag`, it fails:

```py
@cli.command
@cli.add_argument('--flag', type=argparse.FileType('r', encoding='latin-1'))
def read(flag: typing.TextIOWrapper):
    ...
```

It appears to be because it still evaluates the type even if add_argument was used.  I was wondering if there was a reason for this?  In order to achieve a strongly typed variable, I've taken to shadowing variables with proper type annotations to get around this:

```py
@cli.command
@cli.add_argument('--flag', type=argparse.FileType('r', encoding='latin-1'))
def read(sourcefile, flag):
    flag: typing.TextIOWrapper = flag
    ...
```

However, it feels like it would be cleaner if the library just ignored the type annotations for flags where add_argument was used?

**Comment by innot** [comment]
Hi,

this was done to avoid ambiguities and subtle programming errors. Like

```py
@cli.command
@cli.add_argument('foo', type=int)
def test(foo: float):
    ...
```

However I see your point and I think I will introduce a flag to optionally disregard the annotations in the function signature.

Something like this:

```py
cli = ArgParseDecorator(ignore_annotations=True)
```

**Comment by Chris3606** [comment]
Thanks!  One other use case I'm running into where this would be useful is when using this library with a static type checker such as mypy; type checkers in strict mode typically want type annotations on all functions.  Right now, I have to suppress errors from the type checker in these cases; the new flag would help mitigate this as well.

**Comment by innot** [comment]
Just released v1.4.0 with the new ``ignore_annotation`` flag.

This flag is per command (not globally as per my first idea above) which makes sense as the need for using it is probably limited to very few cases.

Usage documentation can be found [here](https://argparsedecorator.readthedocs.io/en/latest/usage.html#ignore-annotations).
